app/components/Layout.tsx

```diff
@@ -1,149 +1,80 @@
-      /* webpackChunkName: "command-bar" */
-      "~/components/CommandBar"
-    )
-);
-
-type Props = WithTranslation &
-  RootStore & {
-    children?: React.ReactNode;
-  };
-
-@observer
-class Layout extends React.Component<Props> {
-  scrollable: HTMLDivElement | null | undefined;
-
-  @observable
-  keyboardShortcutsOpen = false;
-
-  goToSearch = (ev: KeyboardEvent) => {
-    ev.preventDefault();
-    ev.stopPropagation();
-    history.push(searchUrl());
-  };
-
-  goToNewDocument = () => {
-    const { activeCollectionId } = this.props.ui;
-    if (!activeCollectionId) return;
-    const can = this.props.policies.abilities(activeCollectionId);
-    if (!can.update) return;
-    history.push(newDocumentPath(activeCollectionId));
-  };
-
-  render() {
-    const { auth, ui } = this.props;
-    const { user, team } = auth;
-    const showSidebar = auth.authenticated && user && team;
-    const sidebarCollapsed = ui.isEditing || ui.sidebarCollapsed;
-    if (auth.isSuspended) return <ErrorSuspended />;
-
-    return (
-      <Container column auto>
-        <RegisterKeyDown trigger="n" handler={this.goToNewDocument} />
-        <RegisterKeyDown trigger="t" handler={this.goToSearch} />
-        <RegisterKeyDown trigger="/" handler={this.goToSearch} />
-        <RegisterKeyDown
-          trigger="."
-          handler={(event) => {
-            if (isModKey(event)) {
-              ui.toggleCollapsedSidebar();
-            }
-          }}
-        />
-        <Helmet>
-          <title>{team && team.name ? team.name : "Outline"}</title>
-          <meta
-            name="viewport"
-            content="width=device-width, initial-scale=1.0"
-          />
-        </Helmet>
-        <SkipNavLink />
-
-        {this.props.ui.progressBarVisible && <LoadingIndicatorBar />}
-
-        <MobileMenuButton
-          onClick={ui.toggleMobileSidebar}
-          icon={<MenuIcon />}
-          iconColor="currentColor"
-          neutral
-        />
-
-        <Container auto>
-          {showSidebar && (
-            <Switch>
-              <Route path={settingsPath()} component={SettingsSidebar} />
-              <Route component={Sidebar} />
-            </Switch>
-          )}
-
-          <SkipNavContent />
-          <Content
-            auto
-            justify="center"
-            $isResizing={ui.sidebarIsResizing}
-            $sidebarCollapsed={sidebarCollapsed}
-            style={
-              sidebarCollapsed
-                ? undefined
-                : {
-                    marginLeft: `${ui.sidebarWidth}px`,
-                  }
-            }
-          >
-            {this.props.children}
-          </Content>
-
-          <React.Suspense fallback={null}>
-            <Switch>
-              <Route
-                path={`/doc/${slug}/history/:revisionId?`}
-                component={DocumentHistory}
-              />
-            </Switch>
-          </React.Suspense>
-        </Container>
-        <CommandBar />
+
```

Comment:
🍖 This change is effectively a new file with a rename. The old app layout is now `AuthenticatedLayout`.

Reply:
Here we factor out all the display elements of the layout to use in both the authenticated context and the sharing context